src/components/header/Header.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import { NavLink } from 'react-router-dom';
+
+import './header.css';
+
+function Header() {
+
+    const [userName, setUserName] = useState('');
+
+    return (
+        <section id="header">
+            <nav>
+                <div>
+                    <NavLink to="/">Home</NavLink> {' '}
+                    <NavLink to="/income">Income</NavLink> {' '}
+                    <NavLink to="/expenses">Expenses</NavLink> {' '}
+                    <NavLink to="/chart">Chart</NavLink> {' '}
+                    <NavLink to="/login">
```

Comment:
While it is technically possible to have a <button> element as a child of an <a> (hyperlink) element in HTML, it is generally not recommended to do so. The reason is that the `<a>` element is primarily meant to be used for navigation and linking to other resources, while the <button> element is used for triggering actions within a form or interactive element.

The HTML specification states that the content within an <a> element should be text or other inline elements. Placing a <button> element inside an <a> element can lead to confusion for users and potential accessibility issues.

Instead of nesting a <button> element inside an <a> element, it is generally better to choose one or the other based on the intended functionality:

1. If the purpose is to navigate or link to another resource, use only the <a> element.
2. If the purpose is to trigger an action or perform an operation, use only the <button> element.
3. If you need both a link and a button-like appearance, you can style an <a> element to resemble a button using CSS.

Reply:
@noydb why do you recommend this if the NavLink component is provided by the react-router-dom package?

Reply:
He must use the NavLink (as it's an `a` under the hood). I'm just commenting on the actual HTML element that gets rendered. Sorry could have been clearer, I'm not suggesting using an actual `a` directly